src/main/java/org/pentaho/metadata/repository/IMetadataDomainRepositoryDatasourceType.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2022 Hitachi Vantara..  All rights reserved.
+ */
+
+package org.pentaho.metadata.repository;
+
+import java.util.Set;
+
+/**
+ * This interface defines a metadata domain repository, used to maintain a system-wide set of metadata domains.
+ * Includes the ability to retrieve Domains based on data source type.
+ */
+public interface IMetadataDomainRepositoryDatasourceType extends IMetadataDomainRepository {
+
+    /**
+     * Retrieve a list of all the domain ids in the repository of the data source type Metadata.
+     * See {@link #getDomainIds()} for similar functionality.
+     *
+     * @return the metadata domain Ids.
+     */
+    public Set<String> getDomainIdsMetadata();
```

Comment:
Here, you're using a "same prefix" convention w.r.t. `getDomainIds`.

Using the "same suffix" convention would result in `getMetadataDomainIds`, which, I believe, is more like you'd say it in the English natural language.

If these are the short/implicit forms, then there are the long/explicit forms as well!

Using the "same prefix" convention, the long/explicit form could be: `getDomainIds[Of/From]MetadataDataSources` :-)

I cannot think of a long/explitict form for the "same suffix" convention...`getMetadataDataSources.DomainIds` (no way or alternative word to write English's possessive`'` character in a Java identifier!).

I have used the "same prefix" convention extensively in certain scenarios, where a hierarchy of names can be easily understood (e.g. check the names of CCC _options_ in the CCC site https://webdetails.github.io/ccc/ - options documentation at the bottom of the page).

Just my thoughts on the chosen naming.

Reply:
I'm fine with either approach, but `getMetadataDomainIds` sounds more natural . I think I was just making it easier to search all 3 methods by adding the new wording as a suffix.

Reply:
I agree `getMetadataDomainIds `sounds more natural . I think I was just making it easier to search all 3 methods by adding the new wording as a suffix.